services/core-api/app/api/mines/permits/permit_amendment/resources/permit_amendment.py

```diff
@@ -177,6 +183,26 @@ def post(self, mine_guid, permit_guid, permit_amendment_guid=None):
             )
             new_pa.related_documents.append(new_pa_doc)
 
+        if now_application_guid is not None and permit_amendment_status_code == "DFT":
+            application_identity = NOWApplicationIdentity.find_by_guid(now_application_guid)
+            if application_identity.now_application:
```

Comment:
Can we raise `NotFound` if `application_identity.now_application` is `None`?

Reply:
I don't think that's needed, is it? 

If the now_application_guid is not provided that means it creates a permit through the regular flow, its not an error per se

Reply:
@exzizt is this needed?